win/rl/winrl.cc

```diff
@@ -930,6 +930,11 @@ NetHackRL::rl_print_glyph(winid wid, XCHAR_P x, XCHAR_P y, int glyph,
     // No win_proc_calls entry here.
     if (wid == WIN_MAP) {
         instance->store_glyph(x, y, glyph);
+        if (glyph != nul_glyph && color == CLR_BLACK) {
+            /* This will be 'bright black' (or blue) on tty so we change it to
+             * make NLE's colors and tty_colors stay compatible. */
+            color = iflags.wc2_darkgray ? 8 : CLR_BLUE;
```

Comment:
or `CLR_BLUE`?!?

Reply:
I will read the termcap fix

Reply:
NetHack has a configuration option on whether it uses "bright black" (`\033[1;30m`) or (dark) blue to display black entries: https://github.com/facebookresearch/nle/blob/master/win/tty/termcap.c#L887

Reply:
For clarity, here's a Python version of what this code here did before this change:

```python
CLR_BLACK = 0
CLR_RED = 1
CLR_GREEN = 2
CLR_BROWN = 3  # /* on IBM, low-intensity yellow is brown */
CLR_BLUE = 4
CLR_MAGENTA = 5
CLR_CYAN = 6
CLR_GRAY = 7  # /* low-intensity white */
NO_COLOR = 8
CLR_ORANGE = 9
CLR_BRIGHT_GREEN = 10
CLR_YELLOW = 11
CLR_BRIGHT_BLUE = 12
CLR_BRIGHT_MAGENTA = 13
CLR_BRIGHT_CYAN = 14
CLR_WHITE = 15
CLR_MAX = 16

BRIGHT = 8

hilites = [None] * 16

hilites[CLR_GRAY] = ""
hilites[NO_COLOR] = hilites[CLR_GRAY]

for c in range(CLR_MAX // 2):
    if c == CLR_BLACK:
        continue
    hilites[c | BRIGHT] = "\033[1;3%dm" % c
    if c == CLR_GRAY:
        continue
    hilites[c] = "\033[0;3%dm" % c

print(hilites)
```